src/multisig/multisig_clsag_context.cpp

```diff
@@ -0,0 +1,252 @@
+  b_params_L_offset = b_params.size();
+  c_params.resize(c_params.size() + 1);  //this is where L will be inserted later
+  b_params.resize(b_params.size() + num_alpha_components);  //multisig aggregate public nonces for L will be inserted here later
+  c_params_R_offset = c_params.size();
+  b_params_R_offset = b_params.size();
+  c_params.resize(c_params.size() + 1);  //this is where R will be inserted later
+  b_params.resize(b_params.size() + num_alpha_components);  //multisig aggregate public nonces for R will be inserted here later
+  b_params.emplace_back(I);
+  b_params.emplace_back(D);
+  b_params.insert(b_params.end(), s.begin(), s.begin() + l);    //fake responses before 'l'
+  b_params.insert(b_params.end(), s.begin() + l + 1, s.end());  //fake responses after 'l'
+  b_params.emplace_back();
+  encode_int_to_key_le(l, b_params.back());  //real signing index 'l'
+
+  rct::keyV mu_P_params;
+  rct::keyV mu_C_params;
+  mu_P_params.reserve(n * 2 + 4);
+  mu_C_params.reserve(n * 2 + 4);
+
+  mu_P_params.push_back(string_to_key(config::HASH_KEY_CLSAG_AGG_0));
+  mu_C_params.push_back(string_to_key(config::HASH_KEY_CLSAG_AGG_1));
+  mu_P_params.insert(mu_P_params.end(), P.begin(), P.end());
+  mu_C_params.insert(mu_C_params.end(), P.begin(), P.end());
+  mu_P_params.insert(mu_P_params.end(), C_nonzero.begin(), C_nonzero.end());
+  mu_C_params.insert(mu_C_params.end(), C_nonzero.begin(), C_nonzero.end());
+  mu_P_params.emplace_back(I);
+  mu_C_params.emplace_back(I);
+  mu_P_params.emplace_back(scalarmultKey(D, rct::INV_EIGHT));
+  mu_C_params.emplace_back(mu_P_params.back());
+  mu_P_params.emplace_back(C_offset);
+  mu_C_params.emplace_back(C_offset);
+  mu_P = hash_to_scalar(mu_P_params);
+  mu_C = hash_to_scalar(mu_C_params);
+
+  rct::geDsmp I_precomp;
+  rct::geDsmp D_precomp;
+  rct::precomp(I_precomp.k, I);
+  rct::precomp(D_precomp.k, D);
+  rct::key wH_l;
+  rct::addKeys3(wH_l, mu_P, I_precomp.k, mu_C, D_precomp.k);
+  rct::precomp(wH_l_precomp.k, wH_l);
+  W_precomp.resize(n);
+  H_precomp.resize(n);
+  for (std::size_t i = 0; i < n; ++i) {
+    rct::geDsmp P_precomp;
+    rct::geDsmp C_precomp;
+    rct::key C;
+    rct::subKeys(C, C_nonzero[i], C_offset);
+    rct::precomp(P_precomp.k, P[i]);
+    rct::precomp(C_precomp.k, C);
+    rct::key W;
+    rct::addKeys3(W, mu_P, P_precomp.k, mu_C, C_precomp.k);
+    rct::precomp(W_precomp[i].k, W);
+    ge_p3 Hi_p3;
+    rct::hash_to_p3(Hi_p3, P[i]);
+    ge_dsm_precomp(H_precomp[i].k, &Hi_p3);
+  }
+  rct::precomp(G_precomp.k, rct::G);
+  this->l = l;
+  this->s = s;
+  this->num_alpha_components = num_alpha_components;
+
+  initialized = true;
+  return true;
+}
+//----------------------------------------------------------------------------------------------------------------------
+bool CLSAG_context_t::combine_alpha_and_compute_challenge(
+  const rct::keyV& total_alpha_G,
+  const rct::keyV& total_alpha_H,
+  const rct::keyV& alpha,
+  rct::key& alpha_combined,
+  rct::key& c_0,
```

Comment:
The CLSAG code calls this c1. While I honestly kind of prefer c0, and it turns out that's the term from the paper, the discrepancy isn't preferred. We also can't change it within the CLSAG code anymore, which is unfortunate.

Reply:
I agree this isn't ideal... however, `c_0` is set here:
```
for (std::size_t i = (l + 1) % n; i != l; i = (i + 1) % n) {
    if (i == 0)
      c_0 = c;
   ...
}
```
So I think `c_0` makes more sense than doubling down on `c1`.

I added a comment about the notation discrepancy.